### PR TITLE
Minor lazy diagnostic info refactoring

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

**Review by cston — APPROVED** [review]

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1)

**Comment by AlekseyTs** [comment]
@Youssef1313 Thanks for the contribution.

File: src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs [hunk 1]
```diff
@@ -470,6 +470,7 @@ public record [|R|] : IEquatable<R>
     [CompilerGenerated]
     protected R(R original);
 
+    [CompilerGenerated]
```
Comment (Youssef1313) [inline src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs:473]:
```suggestion
```

File: src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs [hunk 2]
```diff
@@ -502,6 +503,16 @@ public record [|R|] : IEquatable<R>
...
+    protected virtual Type EqualityContract
+    {{
+        [CompilerGenerated]
+        get
+        {{
+            return typeof(R);
+        }}
+    }}
+
```
Comment (Youssef1313) [inline src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs:515]:
```suggestion
```

**Review by 333fred — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jcouv @333fred There is an unrelated test failure here

**Comment by 333fred** [comment]
VS_Integration_Debug_64 keeps failing for unrelated reasons. I'm overriding it.

**Pull request merged**